trillian/ctfe/configpb/config.proto

```diff
@@ -132,6 +133,34 @@ message LogConfig {
   // A list of X.509 extension OIDs, in dotted string form (e.g. "2.3.4.5")
   // which should cause submissions to be rejected.
   repeated string reject_extensions = 18;
+
+  // CTFE storage connection string in the following format in general:
+  // driver://[username[:password]@][protocol[(host[:port])]][/[schema|database][?options]]
+  //
+  // MySQL/MariaDB:
+  // mysql://[username[:password]@][protocol[(address)]]/dbname[?param1=value1&...&paramN=valueN]
+  //
+  // This is required when the non-leaf certificate chain storage backend is
+  // CTFE.
+  // 
+  // Warning: CT log operators are advised not to re-use the same connection 
+  // string across multiple LogConfigs due to the log lifecycle.
+  string ctfe_storage_connection_string = 20;
+
+  // An optional list of storage backends for non-leaf certificate chain in 
+  // ExtraData. Log operators may want to use both Trillian GRPC and CTFE as  
+  // the storage backends to ensure the data availability at the beginning. 
+  // By default, the storage backend is Trillian GRPC. To use CTFE as the  
+  // storage backend, the CTFE storage connection string needs to be specified.
+  enum NonLeafCertificateChainStorageBackend {
+    NON_LEAF_CERTIFICATE_CHAIN_STORAGE_BACKEND_TRILLIAN_GRPC = 0;
+    NON_LEAF_CERTIFICATE_CHAIN_STORAGE_BACKEND_CTFE = 1;
+  }
+  repeated NonLeafCertificateChainStorageBackend extra_data_non_leaf_certificate_chain_storage_backends = 21;
+
+  // CTFE non-leaf certificate chain cache.
+  // Setting the cache size (if available in the cache type) avoids the unexpected certificate chain growth rate.
+  Cache extra_data_non_leaf_certificate_chain_cache = 22;
```

Comment:
Did you consider making the cache configured via flags instead of on a per-log config? I  _suspect_ that having this configurable on a per-log basis is overkill. This is also the kind of thing that needs to be tuned, and having these frequent iteration cycles in flags is less dangerous than in this file, which is rarely modified once a log is initialized.

Reply:
Yes, we'll definitely need to have different logical caches per log. This could be done with literally different cache instances, or by using a global cache with some logical namespacing of the entries. e.g. Instead of the key just being the hash of the intermediate cert, it could be `fmt.Sprintf("%s/%x", log.ID, certHash)`

Reply:
The main observation I have about these caching parameters is that they don't affect the _behaviour of the log_, but rather, they affect the _performance/resources of the CTFE instances_ that host the log. It seems reasonable that this would then be controlled by flags, which allows different CTFE instances to be tuned differently. Whether this is for performance benchmarking in a controlled way, or because some machines are bigger than others.

There's also a consideration of orthogonality. Let's say an operator has picked parameters for 5 logs that maximally use the resources available to the CTFE instances hosting them. Now the log operator needs to bring up another log. They would need to edit the log configs of all of the other logs to reduce usage by 20% per log in order to make room for the new log.

Reply:
Removed cache related configs.